### PR TITLE
Add XO Linter and Prettier 

### PR DESCRIPTION
**Pull request by dominicegginton**
Added XO and Prettier with [Lass's](https://lass.js.org) Custom Config to enforce a consistent code styling.

> This PR includes a lot of changes due to refactoring the codebase to follow linting rules

### CODE REVIEW

**Comment by niftylettuce** [comment]
just ping me when ready

**Comment by niftylettuce** [comment]
If you want to bump all the deps lmk - I did bump a bunch of stuff in lass, e.g. remark-preset-github, etc.  See https://github.com/lassjs/lass template

**Comment by dominicegginton** [comment]
@niftylettuce I have bumped dependencies, just need to get some time to sit down and look at refactoring some code to meat  linter code styling. Will ping you when ready

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "singleQuote": true,
+  "bracketSpacing": true,
+  "trailingComma": "none"
+}
```
Comment (IlyaSemenov) [inline .prettierrc:5]: Trailing EOL should always be there; at the very least, missing EOLs should be consistent across the project. (After all, isn't this PR about linting?)

**Comment by niftylettuce** [comment]
@spence-s you could lassify this project

**Review by 3imed-jaberi — COMMENTED** [review]
We will unify the linting and formatting process

**Pull request closed without merging**